src/ui/edittagdialog.cpp

```diff
@@ -842,26 +842,32 @@ void EditTagDialog::FetchTagSongChosen(const Song& original_song,
   const QString filename = original_song.url().toLocalFile();
 
   // Find the song with this filename
-  for (int i = 0; i < data_.count(); ++i) {
-    Data* data = &data_[i];
-    if (data->original_.url().toLocalFile() != filename) continue;
-
-    // Is it currently being displayed in the UI?
-    if (ui_->song_list->currentRow() == i) {
-      // Yes!  We can just set the fields in the UI
-      ui_->title->set_text(new_metadata.title());
-      ui_->artist->set_text(new_metadata.artist());
-      ui_->album->set_text(new_metadata.album());
-      ui_->track->setValue(new_metadata.track());
-      ui_->year->setValue(new_metadata.year());
-    } else {
-      data->current_.set_title(new_metadata.title());
-      data->current_.set_artist(new_metadata.artist());
-      data->current_.set_album(new_metadata.album());
-      data->current_.set_track(new_metadata.track());
-      data->current_.set_year(new_metadata.year());
+  int id;
+  for (id = 0; id < data_.count(); ++id)
+    if (data_[id].original_.url().toLocalFile() == filename)
+      break;
+  
+  if( id == data_.count() ) {
+    qLog(Warning) << "Could not find song to filename: " << filename;
+    return;
+  }
+  
+  Data& data = data_[id];
```

Comment:
Please run `make format` and follow the [style guide](https://google-styleguide.googlecode.com/svn/trunk/cppguide.html)
i.e. `{}` for all `for` and `if` statements, no use of non-const references